File: app/src/main/java/com/lambdaschool/congressdata/OverviewListRepository.java
package com.lambdaschool.congressdata;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.MutableLiveData;
import android.util.Log;

import com.lambdaschool.congressdataapiaccess.CongressDao;
import com.lambdaschool.congressdataapiaccess.CongresspersonOverview;

import java.util.ArrayList;

public class OverviewListRepository {
    public static LiveData<ArrayList<OfficialOverview>> getOverviewList() {
        Log.i("Repository", "Retreiving Data");
        final MutableLiveData<ArrayList<OfficialOverview>> listLiveData = new MutableLiveData<>();

        ArrayList<OfficialOverview>       overviews  = new ArrayList<>();
        ArrayList<CongresspersonOverview> allMembers = CongressDao.getAllMembers();

        for (CongresspersonOverview member:allMembers) {
            overviews.add(new OfficialOverview(member));
        }

        listLiveData.setValue(overviews);

        return listLiveData;
    }
}

File: app/src/main/java/com/lambdaschool/congressdata/OverviewListRepository.kt
package com.lambdaschool.congressdata

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData
import android.util.Log

import com.lambdaschool.congressdataapiaccess.CongressDao
import com.lambdaschool.congressdataapiaccess.CongresspersonOverview

import java.util.ArrayList

object OverviewListRepository {
    val overviewList: LiveData<ArrayList<OfficialOverview>>
        get() {
            Log.i("Repository", "Retreiving Data")
            val listLiveData = MutableLiveData<ArrayList<OfficialOverview>>()

            val overviews = ArrayList<OfficialOverview>()
            val allMembers = CongressDao.getAllMembers()

            for (member in allMembers) {
                overviews.add(OfficialOverview(member))
            }

            listLiveData.value = overviews

            return listLiveData
        }
}
